torch/csrc/jit/backends/backend_interface.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <torch/custom_class.h>
+
+namespace torch {
+namespace jit {
+
+// Interface for a JIT backend.
+class TORCH_API PyTorchBackendInterface : public torch::CustomClassHolder {
```

Comment:
`TORCH_API`  (which is an alias to `CAFFE2_API`) can not be used to implement the code outside of main library, because it becomes `C10_EXPORT` outside of main lib, see https://github.com/pytorch/pytorch/blob/master/c10/macros/Export.h#L96:
```
...
#define C10_HIDDEN
#define C10_EXPORT __declspec(dllexport)
#define C10_IMPORT __declspec(dllimport)
...
#ifdef CAFFE2_BUILD_MAIN_LIB
#define CAFFE2_API C10_EXPORT
#else
#define CAFFE2_API C10_IMPORT
#endif
```
I.e. when `backend_interface.cpp` is compiled as part `torch_python.dll`, `TORCH_API` becomes `__declspec(dllimport)`, which tells linker that vftable should be deinfed in `torch_cpu.dll`, which in turn leads to error you are seeing.

One way to solve the issue is to move `backend_interface.cpp` to `torch_cpu`